File: data/CodeLlama/readable_format_HPEC/fortran/file216.f90
subroutine kernel_jacobi1d(tsteps, n, a, b)
implicit none

DATA_TYPE, dimension(n) :: a
DATA_TYPE, dimension(n) :: b
integer :: n, tsteps
integer :: i, t, j
!$pragma scop
do t = 1, _PB_TSTEPS
do i = 2, _PB_N - 1
b(i) = 0.33333D0 * (a(i - 1) + a(i) + a(i + 1))
end do

do j = 2, _PB_N -1
a(j) = b(j)
end do
end do
!$pragma endscop
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file216.cpp
void kernel_jacobi1d(int tsteps, int n, float* a, float* b)
{
 int i, t, j;
 for (t = 0; t < tsteps; t++)
 {
  for (i = 1; i < n-1; i++)
   b[i] = 0.33333 * (a[i - 1] + a[i] + a[i + 1]);
  for (j = 1; j < n-1; j++)
   a[j] = b[j];
 }
}
